cmd/influxd/run.go

```diff
@@ -42,30 +42,132 @@ func NewRunCommand() *RunCommand {
 }
 
 type Node struct {
-	broker   *influxdb.Broker
-	dataNode *influxdb.Server
+	Broker   *influxdb.Broker
+	DataNode *influxdb.Server
 	raftLog  *raft.Log
+
+	adminServer      *admin.Server
+	clusterListener  net.Listener // The cluster TCP listener
+	snapshotListener net.Listener
 }
 
 func (s *Node) Close() {
-	if s.broker != nil {
-		if err := s.broker.Close(); err != nil {
-			log.Fatalf("error closing broker: %s", err)
+	var b bool
+	if err := s.closeClusterListener(); err != nil {
+		log.Printf("error closing cluster listener: %s", err)
+		b = true
+	}
+
+	if err := s.closeAdminServer(); err != nil {
+		log.Printf("error closing admin server: %s", err)
+		b = true
+	}
+
+	if err := s.closeSnapshotListener(); err != nil {
+		log.Printf("error closing snapshot listener: %s", err)
+		b = true
+	}
+
+	if s.DataNode != nil {
+		if err := s.DataNode.Close(); err != nil {
+			log.Printf("error data broker: %s", err)
+			b = true
+		}
+	}
+
+	if s.Broker != nil {
+		if err := s.Broker.Close(); err != nil {
+			log.Printf("error closing broker: %s", err)
+			b = true
 		}
 	}
 
 	if s.raftLog != nil {
 		if err := s.raftLog.Close(); err != nil {
-			log.Fatalf("error closing raft log: %s", err)
+			log.Printf("error closing raft log: %s", err)
+			b = true
 		}
 	}
 
-	if s.dataNode != nil {
-		if err := s.dataNode.Close(); err != nil {
-			log.Fatalf("error data broker: %s", err)
+	if b {
```

Comment:
Good point.  I wasn't fond of using `log.Fatal` here but we use it in multiple other places in this part of the code instead of returning an error so tried to keep it consistent.  I've updated this to return the first error and the `log.Fatal` is now in the test `Cluster.Close()`